### PR TITLE
cephadm: _parse_ipv6_route: Fix parsing ifs w/o route

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fix parsing interfaces as returned by `ip -6 addr ls` that
don't have a route associated.

Note, in this case there is no route for enp3s0f1

Fixes: https://tracker.ceph.com/issues/53842

Signed-off-by: Sebastian Wagner <sewagner@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by adk3798 — APPROVED** [review]

**Comment by ljflores** [comment]
jenkins test make check

**Comment by ljflores** [comment]
jenkins test make check

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -5662,7 +5662,7 @@ def _parse_ipv6_route(routes: str, ips: str) -> Dict[str, Dict[str, Set[str]]]:
         # find the network it belongs to
         net = [n for n in r.keys()
                if ipaddress.ip_address(ip) in ipaddress.ip_network(n)]
-        if net:
+        if net and iface in r[net[0]]:
             assert(iface)
             r[net[0]][iface].add(ip)
```
[the remark this answers is not in the document]
Comment (sebastian-philipp) [reply src/cephadm/cephadm:5667]: right this is an upgrade test. We can fix this as well, as soon as we backport this to pacific and then set the starting version accordingly.

**Review by ljflores — APPROVED** [review]

**Comment by ljflores** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2022-01-24_16:47:41-orch:cephadm-wip-swagner-testing-2022-01-24-1121-distro-default-smithi/

* failures: ingress

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Pull request merged**